.github/workflows/compareNDandODbuilds.yml

```diff
@@ -2,24 +2,111 @@ name: Build and profile Android apps
 
 on:
   workflow_dispatch:
+    inputs:
+      PULL_REQUEST_NUMBER:
+        description: Pull request number to build the app from
+        required: false
+        default: ''
+      OLD_DOT_PULL_REQUEST_NUMBER:
+        description: Old Dot Pull request number to build the app from
+        required: false
+        default: ''
 
 jobs:
+  getBranchRef:
+    runs-on: ubuntu-latest
+    outputs:
+      REF: ${{ steps.getHeadRef.outputs.REF }}
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+
+      - name: Get ND PR commit SHA
+        id: getHeadRef
+        run: |
+          set -e
+          if [[ -z "${{ github.event.inputs.PULL_REQUEST_NUMBER }}" ]]; then
+            echo "REF=" >> "$GITHUB_OUTPUT"
+          else
+            echo "REF=$(gh pr view ${{ github.event.inputs.PULL_REQUEST_NUMBER }} --json headRefOid --jq '.headRefOid')" >> "$GITHUB_OUTPUT"
+          fi
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+    
+  getOldDotBranchRef:
+    runs-on: ubuntu-latest
+    outputs:
+      OLD_DOT_REF: ${{ steps.getHeadRef.outputs.REF }}
+    steps:
+      - name: Get OD PR commit SHA
+        id: getHeadRef
+        run: |
+          set -e
+          if [[ -z "${{ github.event.inputs.OLD_DOT_PULL_REQUEST_NUMBER }}" ]]; then
+            echo "REF=" >> "$GITHUB_OUTPUT"
+          else
+            echo "REF=$(gh pr view ${{ github.event.inputs.OLD_DOT_PULL_REQUEST_NUMBER }} -R Expensify/Mobile-Expensify --json headRefOid --jq '.headRefOid')" >> "$GITHUB_OUTPUT"
+          fi
+        env:
+          GITHUB_TOKEN: ${{ secrets.OS_BOTIFY_TOKEN }}
+  
+  
+  postGitHubCommentBuildStarted:
```

Comment:
This is now reused in three flows (testBuild, testHybridBuild and this one) could we create a composite action for it?

Reply:
How do you think @mountiny?

Reply:
@jnowakow yeah we dont have to do it here